apps/wpcom-block-editor/src/calypso/iframe-bridge-server.js

```diff
@@ -17,6 +17,63 @@ import tinymce from 'tinymce/tinymce';
  */
 import { inIframe, sendMessage } from './utils';
 
+/**
+ * Monitors Gutenburg for when an editor is opened with content originally authored in the classic editor.
+ *
+ * @param {MessagePort} calypsoPort Port used for communication with parent frame.
+ */
+function triggerConversionPrompt( calypsoPort ) {
+	function addConversionListener( classicBlock ) {
+		// Listen for user response
+		calypsoPort.addEventListener( 'message', onConversionResponse, false );
+		calypsoPort.start();
+
+		function onConversionResponse( message ) {
+			const action = get( message, 'data.action' );
+			const payload = get( message, 'data.payload' );
+
+			if ( action !== 'conversionResponse' ) {
+				return;
+			}
+
+			// Removing the listener regardless of which action was taken
+			calypsoPort.removeEventListener( 'message', onConversionResponse, false );
+
+			// Converting the block if the user selected the "Convert" button in the prompt
+			if ( payload.confirmed === true ) {
+				const unsubscribe = subscribe( () => {
+					dispatch( 'core/editor' ).replaceBlock(
+						classicBlock.clientId,
+						rawHandler( {
+							HTML: classicBlock.originalContent,
+						} )
+					);
+
+					unsubscribe();
+				} );
+			}
+		}
+	}
+
+	// Check if the prompt should open
+	const unsubscribe = subscribe( () => {
+		const { isEditedPostDirty } = select( 'core/editor' );
+		const blocks = select( 'core/editor' ).getBlocks();
+
+		// Open when: editor is not dirty and there is a single classic block
+		if ( ! isEditedPostDirty() && blocks.length === 1 && blocks[ 0 ].name === 'core/freeform' ) {
+			calypsoPort.postMessage( {
+				action: 'openConversions',
+			} );
+
+			// Setup a listener for user response
+			addConversionListener( blocks[ 0 ] );
+
+			unsubscribe();
```

Comment:
Since when we want to show the dialog only once when we open the block editor, we should unsubscribe as soon as the editor is initialized. Otherwise, we will keeping opening the dialog every time the post is dirty and and contains one single classic block.

Perhaps we can use [`getCurrentPost`](https://github.com/WordPress/gutenberg/blob/a8fb376fad486d0eecd8d44358e46ed67755e2f3/packages/editor/src/store/selectors.js#L147-L158) to determine when the editor has been loaded? That way, we can get rid of the `isEditedPostDirty` check.

```js
// Check if the prompt should open
const unsubscribe = subscribe( () => {
	const { getCurrentPost } = select( 'core/editor' );
	const blocks = select( 'core/editor' ).getBlocks();
	const hasInitialized = ! isEmpty( getCurrentPost() );

	if ( ! hasInitialized ) {
		return;
	}
	
	// Make sure we only open the dialog once when the editor loads
	unsubscribe();

	// Open when there is a single classic block
	if ( blocks.length === 1 && blocks[ 0 ].name === 'core/freeform' ) {
		calypsoPort.postMessage( {
			action: 'openConversions',
		} );

		// Setup a listener for user response
		addConversionListener( blocks[ 0 ] );
	}
} );
```

Reply:
AH! Thank you so much! I was wondering how to do that with wp data.